src/main/java/dev/railroadide/railroad/ide/ui/MarkdownPreviewPane.java

```diff
@@ -0,0 +1,204 @@
+            markdownButtons.setVisible(true);
+            showContent(codeView(), topRow);
+        });
+
+        splitView.setOnAction(v -> {
+            markdownButtons.setVisible(true);
+            showContent(splitView(), topRow);
+        });
+
+        previewView.setOnAction(v -> {
+            // hide markdown buttons in preview-only mode
+            markdownButtons.setVisible(false);
+            showContent(previewView(), topRow);
+        });
+
+        return switchButtons;
+    }
+
+    private HBox createMarkdownButtons(){
+        Button headingButton = createButton(FontAwesomeSolid.HEADING);
+
+        var headingMenu = new javafx.scene.control.ContextMenu();
+
+        // example font sizes for H1..H5 (adjust as desired)
+        int[] headingFontSizes = {32, 24, 18, 16, 13, 10};
+
+        for (int level = 1; level <= 6; level++) {
+            CustomMenuItem item = createMenuItem(level, headingFontSizes, headingMenu);
+
+            headingMenu.getItems().add(item);
+        }
+
+        headingButton.setOnAction(e -> {
+            if (!headingMenu.isShowing()) {
+                headingMenu.show(headingButton, javafx.geometry.Side.BOTTOM, 0, 0);
+            } else {
+                headingMenu.hide();
+            }
+        });
+
+
+        Button boldButton = createButton(FontAwesomeSolid.BOLD);
+        setButtonOnAction(boldButton, "**", "**");
+        Button italicButton = createButton(FontAwesomeSolid.ITALIC);
+        setButtonOnAction(italicButton, "_", "_");
+
+        Button quoteButton = createButton(FontAwesomeSolid.QUOTE_LEFT);
+        setButtonOnAction(quoteButton, "> ");
+        Button codeButton = createButton(FontAwesomeSolid.CODE);
+        setButtonOnAction(codeButton, "`", "`");
+        Button linkButton = createButton(FontAwesomeSolid.LINK);
+        setButtonOnAction(linkButton, "[", "](url)");
+
+        Button unorderedListButton = createButton(FontAwesomeSolid.LIST_UL);
+        setButtonOnAction(unorderedListButton, "- ");
+        Button orderedListButton = createButton(FontAwesomeSolid.LIST_OL);
+        setButtonOnAction(orderedListButton, "1. ");
+        //Button taskListButton = createButton(FontAwesomeSolid.TASKS);
+        //setButtonOnAction(taskListButton, "- [ ]");
+
+
+        return new HBox(headingButton, boldButton, italicButton, quoteButton, codeButton, linkButton,
+            unorderedListButton, orderedListButton/*, taskListButton*/);
+    }
+
+    private CustomMenuItem createMenuItem(int level, int[] headingFontSizes, ContextMenu headingMenu) {
+        String labelText = "H" + level;
+
+        javafx.scene.control.Label preview = new javafx.scene.control.Label(labelText);
+        // monospace + padding + approximate size per heading level
+        preview.setStyle("-fx-font-family: 'monospace'; -fx-padding: 6 12; -fx-font-size: " + headingFontSizes[level - 1] + "px;");
+
+        CustomMenuItem item = new CustomMenuItem(preview, true);
```

Comment:
`var` is preferrable here.
```suggestion
        var item = new CustomMenuItem(preview, true);
```